kore/src/Kore/Strategies/Verification.hs

```diff
@@ -377,19 +391,33 @@ transitionRule'
     => [ReachabilityRule]
     -> [Rule ReachabilityRule]
     -> CommonTransitionRule simplifier
-transitionRule' claims axioms = \prim proofState ->
+transitionRule' claims axioms = \prim proofState -> do
+    Foldable.for_ (extractGoalStuck proofState) (\rule -> do
+        let resultPatternPredicate = predicate (getConfiguration rule)
+            multiAndPredicate = getMultiAndPredicate resultPatternPredicate
+        when (any (isNot_Ceil_ . unwrapPredicate) multiAndPredicate) $
+            error . show . Pretty.vsep $
+                [ "Found '\\not(\\ceil(_))' in stuck configuration:"
+                , Pretty.pretty rule
+                , "Please file a bug report."
+                ]
+        )
```

Comment:
Let's extract this to a helper function and include it as one of the transformers below. Be sure to check the _output_ of `transitionRule` and not the input.